core/src/main.cpp

```diff
@@ -629,21 +654,51 @@ auto set_log_level(const json& _config) -> void
 
 auto init_tls(const json& _config) -> void
 {
-	const auto set_env_var = [&](const auto& _json_ptr_path, const char* _env_var, const char* _default_value = "") {
-		using json_ptr = json::json_pointer;
+	// The iRODS connection libraries do not provide a clean way for developers to easily
+	// configure TLS without relying on an irods_environment.json file. All is not lost
+	// though. Turns out the iRODS libraries do inspect environment variables. This gives
+	// the HTTP API a way to hook into the connection logic and support TLS through its
+	// own configuration file. Hence, the following environment-based lambda functions.
 
-		if (const auto v = _config.value(json_ptr{_json_ptr_path}, _default_value); !v.empty()) {
+	const auto set_env_string = [&](const auto& _tls_prop, const char* _env_var, const char* _default_value = "") {
+		const auto element_path = fmt::format("/irods_client/tls/{}", _tls_prop);
+		const auto v = _config.value(json::json_pointer{element_path}, _default_value);
+
+		if (!v.empty()) {
 			const auto env_var_upper = boost::to_upper_copy<std::string>(_env_var);
 			logging::trace("Setting environment variable [{}] to [{}].", env_var_upper, v);
 			setenv(env_var_upper.c_str(), v.c_str(), 1); // NOLINT(concurrency-mt-unsafe)
 		}
 	};
 
-	set_env_var("/irods_client/tls/client_server_policy", irods::KW_CFG_IRODS_CLIENT_SERVER_POLICY, "CS_NEG_REFUSE");
-	set_env_var("/irods_client/tls/ca_certificate_file", irods::KW_CFG_IRODS_SSL_CA_CERTIFICATE_FILE);
-	set_env_var("/irods_client/tls/certificate_chain_file", irods::KW_CFG_IRODS_SSL_CERTIFICATE_CHAIN_FILE);
-	set_env_var("/irods_client/tls/dh_params_file", irods::KW_CFG_IRODS_SSL_DH_PARAMS_FILE);
-	set_env_var("/irods_client/tls/verify_server", irods::KW_CFG_IRODS_SSL_VERIFY_SERVER, "cert");
+	const auto set_env_int = [&_config](const char* _tls_prop, const char* _env_var, int _default_value) {
+		const auto element_path = fmt::format("/irods_client/tls/{}", _tls_prop);
+		const auto v = _config.value(json::json_pointer{element_path}, _default_value);
+
+		const auto env_var_upper = boost::to_upper_copy<std::string>(_env_var);
+		logging::trace("Setting environment variable [{}] to [{}].", env_var_upper, v);
+		const auto v_str = std::to_string(v);
+		setenv(env_var_upper.c_str(), v_str.c_str(), 1); // NOLINT(concurrency-mt-unsafe)
+	};
+
+	// clang-format off
+	set_env_string("client_server_policy", irods::KW_CFG_IRODS_CLIENT_SERVER_POLICY, "CS_NEG_REFUSE");
+	set_env_string("ca_certificate_file", irods::KW_CFG_IRODS_SSL_CA_CERTIFICATE_FILE);
+	set_env_string("verify_server", irods::KW_CFG_IRODS_SSL_VERIFY_SERVER, "cert");
```

Comment:
dh_params_file and certificate_chain_file are missing when compared with the HTTP API implementation. Is that on purpose?